### PR TITLE
Ltac2: Make it possible to define externals with other plugin names

### LINKED ISSUE #16592: Plugins can only add `Ltac2` `external` tactics by pretending to be in coq-core
#### Description of the problem
https://github.com/coq/coq/blob/477c5aac7a92134de0c5660f9b9b71fcd06ed0e8/plugins/ltac2/g_ltac2.mlg#L387-L392
This seems to mean that I need to pretend that all of my tactics are in coq-core.plugins.*

#### Coq Version

8.16

### PR DESCRIPTION
**Pull request by SkySkimmer**
I kept the short name for ltac2 but this can be debated.

Fix #16592

Overlays:
- https://github.com/mit-plv/rewriter/pull/101

### CODE REVIEW

File: doc/sphinx/proof-engine/ltac2.rst [hunk 1]
```diff
@@ -217,9 +217,10 @@ One can define new types with the following commands.
 .. cmd:: Ltac2 @ external @ident : @ltac2_type := @string @string
    :name: Ltac2 external
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:218]:
```suggestion
.. cmd:: Ltac2 @ external @ident : @ltac2_type := @string__plugin @string__function
   :name: Ltac2 external
```

File: doc/sphinx/proof-engine/ltac2.rst [hunk 2]
```diff
@@ -217,9 +217,10 @@ One can define new types with the following commands.
...
 
-   Declares abstract terms.  Frequently, these declare OCaml functions
-   defined in Coq and give their type information.  They can also declare
-   data structures from OCaml.  This command has no use for the end user.
+   Declares functions defined in OCaml. The first :n:`@string` is the
+   plugin name defining the function, the second is the internal name
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:221]:
```suggestion
   Declares functions defined in OCaml. :n:`@string__plugin` is the
   plugin name defining the function.  :n:`@string_function` is the internal name
```

Subscripts are your friend.

File: doc/sphinx/proof-engine/ltac2.rst [hunk 3]
```diff
@@ -217,9 +217,10 @@ One can define new types with the following commands.
...
+   of the function. As a special case, ``"ltac2"`` stands for plugin name
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:222]:
```suggestion
   of the function. As a special case, ``"ltac2"`` represents the plugin name
```

**Review by jfehrle — COMMENTED** [review]
Suggestions

File: plugins/ltac2/g_ltac2.mlg [hunk 4]
```diff
@@ -132,6 +132,10 @@ let opt_fun ?loc args ty e =
   | [] -> e
   | _ :: _ -> CAst.make ?loc (CTacFun (args, e))
 
+let plugin_name = function
+  | "ltac2" -> "coq-core.plugins.ltac2"
```
[the remark this answers is not in the document]
Comment (SkySkimmer) [reply plugins/ltac2/g_ltac2.mlg]: changed

**Review by ppedrot — APPROVED** [review]

File: doc/sphinx/proof-engine/ltac2.rst [hunk 5]
```diff
@@ -214,12 +214,12 @@ One can define new types with the following commands.
...
 
-.. cmd:: Ltac2 @ external @ident : @ltac2_type := @string @string
+.. cmd:: Ltac2 @ external @ident : @ltac2_type := @string__plugin @string__function
    :name: Ltac2 external
 
-   Declares abstract terms.  Frequently, these declare OCaml functions
-   defined in Coq and give their type information.  They can also declare
-   data structures from OCaml.  This command has no use for the end user.
+   Declares functions defined in OCaml. :n:`@string__plugin` is the
+   plugin name defining the function.  :n:`@string_function` is the internal name
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:221]:
```suggestion
   plugin name defining the function.  :n:`@string__function` is the internal name
```

Must be two underscores.

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**